### DISCUSSION TITLE
How to use personal tokens on repos where I am not owner?

### DISCUSSION THREAD

**Discussion by pavel-a** [community/community#53641] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Since we cannot authenticate with passwords any longer and must use the "personal tokens" I cannot push to repos where I have write access, but am not the owner, using the command line git.
And I cannot figure which checkboxes to check so that GH lets me do it. ?????
Tried classic tokens and "fine grained". 
For a "classic" token git returns error 403:  **remote: Write access to repository not granted.**
"Fine grained" even will not let me select repos not **owned** by me!

For example, user N. granted me write access to their repo and **I can push with SmartGit** (uses OAUTH).

But I cannot figure how to make GH accept my username and a token instead of password???
I don't want to install a GUI client in every VM or container.
Please advice.

**Comment by airtower-luna** [comment, marked as answer]
> For a "classic" token git returns error 403: remote: Write access to repository not granted.

Which scopes did you enable when creating the token? Generally you should need **repo** (for repository access, like pushing) and if you use Actions also **workflow** (required to modify workflow files).

**Reply by pavel-a** [reply]
Of course, you're right. With "repo" checked (but all sub-items unchecked) it gives the access. Thank you,

Now, is it possible to let users create "fine grained" tokens **for a specific repos, owned by somebody else**?
Or we have to ask the repo owner every time?

**Reply by airtower-luna** [reply]
According to the documentation, organization admins can opt-in: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#fine-grained-personal-access-tokens

I can't see any way to create one for another user's private repo that you're a collaborator on.

**Reply by yedayak** [reply]
This is on github's public roadmap: https://github.com/github/roadmap/issues/601 FWIW

**Reply by ben-foxmoore** [reply]
> This is on github's public roadmap: [github/roadmap#601](https://github.com/github/roadmap/issues/601) FWIW

As GA for their Enterprise product, which makes no sense given their own feature description says 
> This ensures that open source projects [...] can still use fine-grained PATs

What open source projects are using Github Enterprise...

**Comment by Y-T-G** [comment]
The correct answer is to add the SSH key from the system to your account and then clone the repo using SSH.

**Reply by pdpino** [reply]
This is a possible solution, but SSH keys grant even more permissions. If I'm simply trying to clone a read-only repository in a server shared with other users, I might not want to setup an ssh-key with full access to my github account (even if I save the ssh-key with a passphrase).
